core/src/main/scala/org/apache/spark/sql/TiStrategy.scala

```diff
@@ -137,47 +139,54 @@ class TiStrategy(context: SQLContext) extends Strategy with Logging {
         throw new IllegalArgumentException("Should never be here")
 
       case f @ Sum(BasicExpression(arg)) =>
-        dagRequest.addAggregate(new TiSum(arg), fromSparkType(f.dataType))
+        dagRequest
+          .addAggregate(AggregateFunction.newCall(FunctionType.Sum, arg), fromSparkType(f.dataType))
 
-      case f @ Count(args) =>
+      case f @ Count(args) if args.length == 1 =>
```

Comment:
Why we always have `args.length == 1` here?

Reply:
Seems `count(any_thing)` has been rewritten to `count(1)` in `lineitem`, but why is this happen?

Reply:
Only if use with count(distinct) you can have more than one columns here, but such cases cannot be pushed down. So we do only push count(column)
count(1) transform only happen when column is not null. In that case, count(1) equals to count(column)

Reply:
Checked one more time it seems multiple columns without distinct means + 1 if any column is not null.